modules/evm/src/lib.rs

```diff
@@ -1363,18 +1367,29 @@ impl<T: Config> Pallet<T> {
 				}
 			});
 
-			ContractStorageSizes::<T>::take(contract);
+			let _total_size = ContractStorageSizes::<T>::take(contract);
 
-			T::IdleScheduler::schedule(
+			// schedule to remove
+			task_id = T::IdleScheduler::schedule(
 				EvmTask::Remove {
 					caller: *caller,
 					contract: *contract,
 					maintainer: contract_info.maintainer,
 				}
 				.into(),
-			)
+			)?;
+
+			Ok(())
 		})?;
 
+		// try to dispatch the task, limit 50 DB writes
+		let weight_limit = Weight::from_ref_time(
+			<T as frame_system::Config>::DbWeight::get()
+				.write
+				.saturating_mul(REMOVE_LIMIT.saturating_div(2).into()),
```

Comment:
I think it is better to just have two constant. This could be `IMMEDIATE_REMOVE_LIMIT`